### DISCUSSION TITLE
Is there a plan to release a stable 2.x version ?

### DISCUSSION THREAD

**Discussion by NinjaCross** [MichaCo/CacheManager#355] [General]
My customers are worried of using beta-grade packages.
As far as I can see, the version 2.0.0 is in beta stage since the 11/15/2018
Is there a plan to release a 2.0.0 non-beta ?

**Comment by MichaConrad** [comment]
yeah if you look at other issues and discussions and the open PR you'll see its in the works and I'm almost done with it.
I just have very limited time right now because of work. But will be release soon™

**Reply by NinjaCross** [reply]
That's a great news !

**Reply by NinjaCross** [reply]
Hi @MichaCo 
Any news about the release of the 2.x version ?
Thanks again for your support.

**Reply by NinjaCross** [reply]
Hi @MichaCo .
Sorry to bother, but are there news about this ?

**Comment by tonyqus** [comment]
Would you please let me know the release notes/bug list of 2.0? I'm going to fork this project and contribute but I don't know where to start.

I found a branch called work/prep-release and there is a pending PR there. Is that for 2.0 release?

**Reply by jerviscui** [reply]
How's this topic going?

**Reply by tonyqus** [reply]
No reply from the author. Looks he doesn't have time to contribue anymore but he also doesn't wanna anyone to pick up.  Anyway, I totally understand how hard it is to maintain a profitless project.

**Reply by MichaConrad** [reply]
I might have some time soon to finally look at this again and make a release.
I just didn't want to say something here because that would just create certain expectations... Anyways

I'm sorry if that's maybe not what you want to hear ;) 
but, lets see

**Reply by tonyqus** [reply]
@MichaCo  It's nice to see your reply. If you don't have time, you can somewhat change your role in this project. I see a few guys do project management of their open source projects instead of coding by himself. From project perspective, if a project wanna survive, it must be kept maintaining. No matter who is the coding guy.

**Comment by Mek7** [comment]
I am currently migrating from .NET Framework to .NET 8 and one of my class libraries references CacheManager nuget packages of version 1.2.0. These don't target .NET Standard 2.0 so I get build warnings that they may be not compatible with my project (and there could be runtime errors).
A 2.x release of CacheManager should solve this. Seeing in the readme of CacheManager that .NET Standard 2.0 is supported... it looks like the documentation is out of sync with the actual release of CacheManager. Please release CacheManager 2.x on nuget so that .NET Standard 2.0 is supported. Thank you.

**Reply by xXAvoraXx** [reply]
Is there any update? or new version?

**Reply by MichaConrad** [reply]
I would actually really like to release a new version. My main problem is that I do not want to re-test every single cache system. 
I personally only use Redis. There are now also alternatives for the use cases I built with CacheManager. Redis basically has that mutli layer cache stuff built in now and there is even a new package from Microsoft for distributed + memory cache.

So, yeah, idk how valuable a new version of this package really is.
But I'm probably just going to release a version and only test the Redis part. I could also just remove memcache and all the other stuff noone is probably using anyways ;)

**Reply by xXAvoraXx** [reply]
On the Microsoft side, they released Garnet as an alternative to Redis, did you have a chance to try it?

https://microsoft.github.io/garnet/docs

https://microsoft.github.io/garnet/docs/benchmarking/results-resp-bench

**Reply by MichaConrad** [reply]
Nope, haven't tried that. But it sounds like it should just work with CacheManager because it is basically using the Redis protocol. Unless there are features missing which I need it should just work (or maybe with setting proxy mode on, idk) 

> Garnet uses Redis' RESP as its primary wire protocol. Thus, one can use Garnet with unmodified Redis clients [available](https://redis.io/docs/latest/develop/connect/clients/) in most programming languages, for example, with [StackExchange.Redis](https://github.com/StackExchange/StackExchange.Redis)

I have no plans to test this though.

**Reply by MichaConrad** [reply]
@xXAvoraXx I'm actually using Garnet for unit tests now, seems to work pretty well.
Main painpoint for anyone using Cachemanager is the lack of support for key space notifications though. CacheManager is using those to invalid the local cache in case the element got deleted from Redis without using CacheManager.  see https://github.com/microsoft/garnet/issues/915

**Comment by Mek7** [comment]
@MichaCo Redis-only support is just fine. Our application uses it in combination with system memory to not call Redis so often, would this feature be preserved in the new version?
We are in the process of thinking about a new cache library, since CacheManager 1 is so old...

**Reply by MichaConrad** [reply]
Yea that would still work

**Comment by sberwal** [comment]
Hi Micha

Looking forward to the release of latest version of CacheManager. We use it inside Ocelot and we believe the new version with Redis support could be useful.

Cheers

**Comment by MichaConrad** [comment]
I have a new V2 version out, mostly all things updated, but as mentioned, with some breaking changes.

There is again only a beta package on Nuget, because I want to test it at least a little bit against real code tomorrow.

But this should go live in the next few days.



## Version 2.0 Breaking Changes

* CacheManager.Microsoft.Extensions.Logging is not a separated package anymore. Logging is now part of the Core package.
* CacheManager.Memcached is not supported anymore
* CacheManager.Couchbase is not supported anymore
* CacheManager.Web is not supported anymore

### Testing with Microsoft.Garnet

For testing and benchmarking, this project is now using Microsoft.Garnet, which allows to create a Redis server which runs in process and is easy to setup.
This has some limitations though and before you use Microsoft.Garnet in production, be aware that for example key space notifications are not supported yet.

**Reply by tonyqus** [reply]
May I ask when will 2.0 release?

**Reply by MichaConrad** [reply]
If this was a joke, good one 😃 

Otherwise
![image](https://github.com/user-attachments/assets/78c8e570-22c2-4106-8ec6-c4fe6e8f8baf)

**Reply by tonyqus** [reply]
Sorry, I missed some text. 

Can you also update the release in this repo? https://github.com/MichaCo/CacheManager/releases

And create a new tag for it. Usually I create rc1,rc2 tag for NPOI release

**Reply by MichaConrad** [reply]
yea sure, I will do that when its released

**Reply by tonyqus** [reply]
It's nice to see the 2.0 release after 2 years waiting.

**Reply by sberwal** [reply]
Hi @MichaCo 

that is a welcome news. I hope we can use redis directly without Microsoft.garnet as well if required in this release?

**Reply by MichaConrad** [reply]
Ofc, you can use whatever server you want ;) 

I'm just using Microsoft.Garnet to test because it is easier to setup and run than using Redis on Linux or something, and I can run it on the build agent in Azure Pipelines without problems.

**Comment by MichaConrad** [comment]
V2 is released
